03_python_fastapi_project/main.py

```
@@ -113,7 +128,91 @@ async def delete_product(product_id: int, db: AsyncSession = Depends(get_db)):
+    result = await db.execute(select(Product).filter(Product.id == db_cart_item.product_id))
+    product = result.scalar_one_or_none()
+
+    if product:
+        product.stock += db_cart_item.quantity
+
+    await db.delete(db_cart_item)
+    await db.commit()
+    return {"message": "Item removed from cart"}
+
+@app.put("/cart/items/{item_id}", response_model=CartItemResponse)
+async def update_cart_item(item_id: int, quantity: int, db: AsyncSession = Depends(get_db)):
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)